geopandas/geodataframe.py

```diff
@@ -1419,6 +1419,16 @@ def __getitem__(self, key):
         return a GeoDataFrame.
         """
         result = super().__getitem__(key)
+        # Custom logic to avoid waiting for pandas GH51895
+        # result is not geometry dtype for multi-indexes
+        if (
+            isinstance(result, Series)
+            and isinstance(self.columns, pd.MultiIndex)
+            and not is_geometry_type(result)
```

Comment:
Do we only need to do this when we are getting the geometry column? (based on checking `self._geometry_column_name`?) Because now we are getting the column twice for every other non-geometry column in case of a MultiIndex columns?

Reply:
As far as I can tell, we can't do this specifically for `self._geometry_column_name` because we're catching an internal call of `_getitem_multilevel` where `key=""` checking that seems to be the next best thing. I'm hoping that `is_scalar` is safe to use and isn't depending on internals. I'm also not certain if `isinstance(result, Series)` is required, but doesn't seem straightforward to know if for certain this code path will never give a dataframe.

I've updated to squeeze(axis="columns") below, I'm not sure if that's what you had in mind.